### ISSUE TITLE
uint32_t  undefined  identifier error on Visual studio 2013 with boost 1.58

### ISSUE THREAD

**Issue by cwyyprog** [bitshares/bitshares-core#1296]
I am trying to build the repository using Appveyor's Visual Studio 2013 environment for #964. Other environment like Visual Studio 2015 from Appveyor will certainly can not build due to #389. Currently, my endeavors on  are Appveyor's Visual Studio 2013 environment blocked by an undefined identifier error of uint32_t in library/fc submodule. 
```
"C:\projects\bitshares-core\bitshares-core\install.vcxproj" (default target) (1) ->
"C:\projects\bitshares-core\bitshares-core\ALL_BUILD.vcxproj" (default target) (3) ->
"C:\projects\bitshares-core\bitshares-core\libraries\fc\tests\all_tests.vcxproj" (default target) (4) ->
"C:\projects\bitshares-core\bitshares-core\libraries\fc\fc.vcxproj" (default target) (5) ->
(ClCompile target) -> 
  C:\projects\bitshares-core\bitshares-core\libraries\fc\src\asio.cpp(132): error C2462: 'fc::asio::default_io_service_scope::{ctor}::boost_se_params_t_19' : cannot define a type in a 'new-expression' [C:\projects\bitshares-core\bitshares-core\libraries\fc\fc.vcxproj]
  C:\projects\bitshares-core\bitshares-core\libraries\fc\src\asio.cpp(132): error C2462: 'fc::asio::default_io_service_scope::{ctor}::boost_se_guard_t_19' : cannot define a type in a 'new-expression' [C:\projects\bitshares-core\bitshares-core\libraries\fc\fc.vcxproj]
  C:\projects\bitshares-core\bitshares-core\libraries\fc\include\fc/container/flat_fwd.hpp(15): error C2061: syntax error : identifier 'uint32_t' [C:\projects\bitshares-core\bitshares-core\libraries\fc\fc.vcxproj]
  C:\projects\bitshares-core\bitshares-core\libraries\fc\include\fc/container/flat_fwd.hpp(17): error C2061: syntax error : identifier 'uint32_t' [C:\projects\bitshares-core\bitshares-core\libraries\fc\fc.vcxproj]
  C:\projects\bitshares-core\bitshares-core\libraries\fc\include\fc/container/flat_fwd.hpp(19): error C2061: syntax error : identifier 'uint32_t' [C:\projects\bitshares-core\bitshares-core\libraries\fc\fc.vcxproj]
  C:\projects\bitshares-core\bitshares-core\libraries\fc\include\fc/container/flat_fwd.hpp(21): error C2061: syntax error : identifier 'uint32_t' [C:\projects\bitshares-core\bitshares-core\libraries\fc\fc.vcxproj]
  C:\projects\bitshares-core\bitshares-core\libraries\fc\include\fc/container/flat_fwd.hpp(25): error C2061: syntax error : identifier 'uint32_t' [C:\projects\bitshares-core\bitshares-core\libraries\fc\fc.vcxproj]
  C:\projects\bitshares-core\bitshares-core\libraries\fc\include\fc/container/flat_fwd.hpp(27): error C2061: syntax error : identifier 'uint32_t' [C:\projects\bitshares-core\bitshares-core\libraries\fc\fc.vcxproj]
  C:\projects\bitshares-core\bitshares-core\libraries\fc\include\fc/container/deque_fwd.hpp(10): error C2061: syntax error : identifier 'uint32_t' [C:\projects\bitshares-core\bitshares-core\libraries\fc\fc.vcxproj]
  C:\projects\bitshares-core\bitshares-core\libraries\fc\include\fc/container/deque_fwd.hpp(12): error C2061: syntax error : identifier 'uint32_t' [C:\projects\bitshares-core\bitshares-core\libraries\fc\fc.vcxproj]
    396 Warning(s)
    10 Error(s)
```
I have build everything succesfully except the clitest (see #1292) using Visual Studio 2015 update 1 on my laptop.

**Comment by pmconrad** [comment]
Please try adding <cstdint> to the list of includes.

**Comment by pmconrad** [comment]
#include \<algorithm>
#include \<thread>
**#include \<cstdint>**

namespace fc {

**Comment by ryanRfox** [comment]
@cwyyprog Please let us know if this resolved your build issue. If yes, please close the Issue. If no, please provide additional details you are currently encountering. Please note: Core Team may close this issue for inactivity in the future. Thanks

**Comment by abitmore** [comment]
Hi @cwyyprog, do we need a PR to fix this?

**Comment by cwyyprog** [comment]
No, the issue is not fixed. I checked out the master branch and add this PR on my local repository, and another issue happened, this issue disappears, so I closed this issue, moved to the other issue. To answer @abitmore 's question, I tried to compile the bitshares-fc  library separately with this PR, and it turned out the issue remains unfixed: 
```
 "C:\Users\800xa\bitshares-core\libraries\fc\all_build.vcxproj" (default
 target) (1) ->
 "C:\Users\800xa\bitshares-core\libraries\fc\tests\blind.vcxproj" (defaul
 t target) (3) ->
 "C:\Users\800xa\bitshares-core\libraries\fc\fc.vcxproj" (default target)
  (4) ->
 (ClCompile target) ->
   C:\Users\800xa\bitshares-core\libraries\fc\src\asio.cpp(133): error C2
 462: 'fc::asio::default_io_service_scope::{ctor}::boost_se_params_t_19'
 : cannot define a type in a 'new-expression' [C:\Users\800xa\bitshares-c
 ore\libraries\fc\fc.vcxproj]
   C:\Users\800xa\bitshares-core\libraries\fc\src\asio.cpp(133): error C2
 462: 'fc::asio::default_io_service_scope::{ctor}::boost_se_guard_t_19' :
  cannot define a type in a 'new-expression' [C:\Users\800xa\bitshares-co
 re\libraries\fc\fc.vcxproj]
   C:\Users\800xa\bitshares-core\libraries\fc\include\fc/container/flat_f
 wd.hpp(15): error C2061: syntax error : identifier 'uint32_t' [C:\Users\
 800xa\bitshares-core\libraries\fc\fc.vcxproj]
   C:\Users\800xa\bitshares-core\libraries\fc\include\fc/container/flat_f
 wd.hpp(17): error C2061: syntax error : identifier 'uint32_t' [C:\Users\
 800xa\bitshares-core\libraries\fc\fc.vcxproj]
   C:\Users\800xa\bitshares-core\libraries\fc\include\fc/container/flat_f
 wd.hpp(19): error C2061: syntax error : identifier 'uint32_t' [C:\Users\
 800xa\bitshares-core\libraries\fc\fc.vcxproj]
   C:\Users\800xa\bitshares-core\libraries\fc\include\fc/container/flat_f
 wd.hpp(21): error C2061: syntax error : identifier 'uint32_t' [C:\Users\
 800xa\bitshares-core\libraries\fc\fc.vcxproj]
   C:\Users\800xa\bitshares-core\libraries\fc\include\fc/container/flat_f
 wd.hpp(25): error C2061: syntax error : identifier 'uint32_t' [C:\Users\
 800xa\bitshares-core\libraries\fc\fc.vcxproj]
   C:\Users\800xa\bitshares-core\libraries\fc\include\fc/container/flat_f
 wd.hpp(27): error C2061: syntax error : identifier 'uint32_t' [C:\Users\
 800xa\bitshares-core\libraries\fc\fc.vcxproj]
   C:\Users\800xa\bitshares-core\libraries\fc\include\fc/container/deque_
 fwd.hpp(10): error C2061: syntax error : identifier 'uint32_t' [C:\Users
 \800xa\bitshares-core\libraries\fc\fc.vcxproj]
   C:\Users\800xa\bitshares-core\libraries\fc\include\fc/container/deque_
 fwd.hpp(12): error C2061: syntax error : identifier 'uint32_t' [C:\Users
 \800xa\bitshares-core\libraries\fc\fc.vcxproj]
```

**Comment by pmconrad** [comment]
Closing since we have decided to move to C++14.
